packages/wizards-of-react/src/components/contentNodes/SelectDatePicker.tsx

```diff
@@ -112,25 +113,23 @@ export const SelectDatePicker: React.FC<TSelectDatePickerProps> = ({
       </Select>
       <Select
         value={day ?? ""}
-        disabled={disabled || monthIndex == null}
+        disabled={disabled || month == null}
         size={size}
         onChange={(e) => {
           const newDay = Number(e.target.value);
           setDay(newDay);
-          propagateChange(year, monthIndex, newDay);
+          propagateChange(year, month, newDay);
         }}
       >
         <option value="">-- Day --</option>
-        {Array(monthIndex !== undefined ? getDaysInMonth(new Date(year ?? endYear, monthIndex ?? 0)) : 0)
+        {Array(month !== undefined ? getDaysInMonth(new Date(year ?? endYear, month ?? 0)) : 0)
```

Comment:
nit - we could clean up /refacto this line to make this easier to read

Reply:
yeah was trying to figure out where to draw the line myself on refactoring the component, it's definitely a bit hard to grok currently. Think the next level for me would be a partial rewrite of the whole component, and IMO it's not quite at that level yet